src/api/cvc4cpp.cpp

```diff
@@ -2023,9 +2023,19 @@ Term Solver::mkBVFromStrHelper(uint32_t size,
       << "base 2, 10, or 16";
 
   Integer val(s, base);
-  CVC4_API_CHECK(val.modByPow2(size) == val)
-      << "Overflow in bitvector construction (specified bitvector size " << size
-      << " too small to hold value " << s << ")";
+
+  if (val.strictlyNegative())
+  {
+    CVC4_API_CHECK(val >= -Integer("2", 10).pow(size - 1))
```

Comment:
The condition for `CVC4_API_CHECK` is the one you want to hold for the check to *not* fail -- looks like this condition is the fail condition? Also, I'm missing a corresponding unit test.

Reply:
Oops, yeah forgot to add unit tests. Added with e7193b6d28ff143693267d85ad610b4c15dd08d2. Regarding the API check, it's correct as is. The value is negative so it needs to be greater than the minimum bound. It's tested in the added unit test.

Reply:
Ah right, my bad.